File: 01gpio/02digital_out/digital_out_wpi.js
var wpi = require('wiring-pi');
var readline = require('readline');

var rl = readline.createInterface(process.stdin, process.stdout);

var pin = 18;
wpi.setup('gpio');
wpi.pinMode(pin, wpi.OUTPUT);

rl.setPrompt("1:On, 0:Off >");

rl.prompt();
rl.on('line', function(answer){
	wpi.digitalWrite(pin, Number(answer));
	rl.prompt();
});

File: 01gpio/02digital_out/digital_out_wpi.py
import wiringpi as wpi

PIN = 18
wpi.wiringPiSetupGpio()
wpi.pinMode(PIN, wpi.OUTPUT)
print(f'#{PIN} is set to OUTPUT.')
try:
    while True:
        val = input("1:on, 0:off > ")
        if val == '0':
            wpi.digitalWrite(PIN, wpi.LOW)
            print('On')
        elif val == '1':
            wpi.digitalWrite(PIN, wpi.HIGH)
            print('Off')
        else:
            break
finally:
    wpi.digitalWrite(PIN, wpi.LOW)
    wpi.pinMode(PIN, wpi.INPUT)
    print('cleanup and exit')

